fastly/block_fastly_service_v1_director.go

```diff
@@ -106,6 +117,84 @@ func (h *DirectorServiceAttributeHandler) Process(d *schema.ResourceData, latest
 			}
 		}
 	}
+
+	// UPDATE modified resources
+	//
+	// NOTE: although the go-fastly API client enables updating of a resource by
+	// its 'name' attribute, this isn't possible within terraform due to
+	// constraints in the data model/schema of the resources not having a uid.
+	for _, resource := range diffResult.Modified {
+		resource := resource.(map[string]interface{})
+
+		opts := gofastly.UpdateDirectorInput{
+			ServiceID:      d.Id(),
+			ServiceVersion: latestVersion,
+			Name:           resource["name"].(string),
+		}
+
+		// only attempt to update attributes that have changed
+		modified := setDiff.Filter(resource, oldSet)
+
+		// NOTE: where we transition between interface{} we lose the ability to
+		// infer the underlying type being either a uint vs an int. This
+		// materializes as a panic (yay) and so it's only at runtime we discover
+		// this and so we've updated the below code to convert the type asserted
+		// int into a uint before passing the value to gofastly.Uint().
+		if v, ok := modified["comment"]; ok {
+			opts.Comment = gofastly.String(v.(string))
+		}
+		if v, ok := modified["shield"]; ok {
+			opts.Shield = gofastly.String(v.(string))
+		}
+		if v, ok := modified["quorum"]; ok {
+			opts.Quorum = gofastly.Uint(uint(v.(int)))
+		}
+		if v, ok := modified["type"]; ok {
+			switch v.(int) {
+			case 1:
+				opts.Type = gofastly.DirectorTypeRandom
+			case 2:
+				opts.Type = gofastly.DirectorTypeRoundRobin
+			case 3:
+				opts.Type = gofastly.DirectorTypeHash
+			case 4:
+				opts.Type = gofastly.DirectorTypeClient
+			}
+		}
+		if v, ok := modified["retries"]; ok {
+			opts.Retries = gofastly.Uint(uint(v.(int)))
+		}
+		if v, ok := modified["capacity"]; ok {
+			opts.Capacity = gofastly.Uint(uint(v.(int)))
+		}
+
+		log.Printf("[DEBUG] Update Director Opts: %#v", opts)
+		_, err := conn.UpdateDirector(&opts)
+		if err != nil {
+			return err
+		}
+
+		if v, ok := modified["backends"]; ok {
+			backends := v.(*schema.Set).List()
+			if len(backends) > 0 {
+				for _, backend := range backends {
+					opts := gofastly.CreateDirectorBackendInput{
```

Comment:
@phamann I created a go-fastly script to validate what happens if we try to create a backend that already exists, and it seems we'll get a `409 Conflict` error.

One way to avoid any errors is to ignore 409's but because go-fastly currently doesn't return any useful contextual information about the error that came back from the API (note: I have made a ticket for this annoyance) I would need to do something like:

```go
// Ignore a 409 Conflict
if !strings.Contains(err.Error(), "409 - Conflict") {
    return err
}
```

Reply:
I should add, the tests appear to update a backend's name and it's picked up as 'modified' so it looks like we wouldn't end up trying to recreate existing director backends.